salt/modules/archive.py

```diff
@@ -157,7 +158,10 @@ def _list_tar(name, cached, decompress_cmd, failhard=False):
         files = []
         links = []
         try:
-            with contextlib.closing(tarfile.open(cached)) as tar_archive:
+            open_kwargs = {'name': cached} \
+                if not isinstance(cached, subprocess.Popen) \
+                else {'fileobj': cached.stdout, 'mode': 'r|*'}
```

Comment:
You can actually drop the * here.  The subprocess should be returning an uncompressed tar archive, so there's no need to run the decompression heuristic on it.